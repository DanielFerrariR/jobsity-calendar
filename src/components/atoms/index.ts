export { default as Box } from './box'
export { default as Button } from './button'
export { default as Paper } from './paper'
export { default as Typography } from './typography'
export { default as TextField } from './text_field'
export { default as Container } from './container'
export { default as AppBar } from './appbar'
export { default as IconButton } from './icon_button'
export { default as Toolbar } from './toolbar'
export { default as Image } from './image'
export { default as Grid } from './grid'
export { default as Link } from './link'
export { default as Checkbox } from './checkbox'
export { default as Accordion } from './accordion'
export { default as AccordionSummary } from './accordion_summary'
export { default as AccordionDetails } from './accordion_details'
export { default as Tooltip } from './tooltip'
export { default as Modal } from './modal'
export { default as MenuItem } from './menu_item'
export { default as ListItemIcon } from './list_item_icon'
export { default as ListItemText } from './list_item_text'
export { default as ListItemSecondaryAction } from './list_item_secondary_action'
export { default as MenuList } from './menu_list'
export { default as Popover } from './popover'
export { default as Fade } from './fade'
export { default as Backdrop } from './backdrop'
export { default as CircularProgress } from './circular_progress'
export { default as LinearProgress } from './linear_progress'
export { default as Skeleton } from './skeleton'
export { default as List } from './list'
export { default as ListItem } from './list_item'
export { default as ClickAwayListener } from './click_away_listener'
export { default as Badge } from './badge'
export { default as Portal } from './portal'
export { default as Snackbar } from './snackbar'
export { default as Step } from './step'
export { default as StepContent } from './stepContent'
export { default as StepLabel } from './stepLabel'
export { default as Stepper } from './stepper'
export { default as TableBody } from './table_body'
export { default as TableContainer } from './table_container'
export { default as TableHead } from './table_head'
export { default as TableRow } from './table_row'
export { default as Table } from './table'
export { default as TableCell } from './table_cell'
export { default as DateTimePicker } from './date_time_picker'
