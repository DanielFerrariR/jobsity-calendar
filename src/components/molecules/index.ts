export { CustomTableHead, CustomTableBody } from './custom_table'
export { default as Modal } from './modal'
export { default as ReminderCard } from './reminder_card'
