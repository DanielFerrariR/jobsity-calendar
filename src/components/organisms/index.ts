export { default as Calendar } from './calendar'
export { default as ListRemindersModal } from './list_reminders_modal'
export { default as CreateReminderModal } from './create_reminder_modal'
export { default as EditReminderModal } from './edit_reminder_modal'
