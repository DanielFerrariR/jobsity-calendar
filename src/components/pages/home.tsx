import React from 'react'
import { Container } from 'src/components/atoms'
import { Calendar } from 'src/components/organisms'

const Home: React.FC = () => {
  return (
    <Container mt={4}>
      <Calendar />
    </Container>
  )
}

export default Home
