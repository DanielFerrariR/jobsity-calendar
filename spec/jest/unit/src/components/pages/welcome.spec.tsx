import React from 'react'
import { Welcome } from 'src/components/pages'
import { render } from 'spec/jest/config/test_utils'
import { axe } from 'jest-axe'

describe('testing Welcome component', () => {
  test('should snapshot works with english language', () => {
    const { container } = render(<Welcome />)

    expect(container).toMatchSnapshot()
  })

  test('should snapshot works with portuguese language', () => {
    const { container } = render(<Welcome />, {
      language: 'pt'
    })

    expect(container).toMatchSnapshot()
  })

  test('should snapshot works with spanish language', () => {
    const { container } = render(<Welcome />, {
      language: 'es'
    })

    expect(container).toMatchSnapshot()
  })

  test('should the page be accessible', async () => {
    const { container } = render(<Welcome />)
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
