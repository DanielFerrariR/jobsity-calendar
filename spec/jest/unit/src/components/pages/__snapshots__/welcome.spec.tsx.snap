// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Welcome component should snapshot works with english language 1`] = `
<div>
  <div
    class="MuiContainer-root Component-root Component-root MuiContainer-maxWidthMd"
  >
    <img
      alt="Welcome logo"
      class="img-root img-root"
      src=""
    />
    <div
      class="MuiGrid-root WithStyles(ForwardRef(Grid))-root WithStyles(ForwardRef(Grid))-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root WithStyles(ForwardRef(Grid))-root WithStyles(ForwardRef(Grid))-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained Component-root Component-root MuiButton-containedSecondary"
          data-testid="register-button"
          href="/front/register"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            Create account
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
      <div
        class="MuiGrid-root WithStyles(ForwardRef(Grid))-root WithStyles(ForwardRef(Grid))-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained Component-root Component-root MuiButton-containedPrimary MuiButton-fullWidth"
          data-testid="login-button"
          href="/front/login"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            Sign in
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
    <div
      class="MuiPaper-root Component-root Component-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <p
        class="MuiTypography-root Component-root Component-root MuiTypography-body1 MuiTypography-alignCenter"
        data-testid="welcome-board-title"
      >
        If you already have an account
      </p>
      <p
        class="MuiTypography-root Component-root Component-root MuiTypography-body1 MuiTypography-alignCenter"
        data-testid="welcome-board-subtitle"
      >
        Sign in
      </p>
      <img
        alt="Welcome interface"
        class="img-root img-root"
        src=""
      />
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root Component-root Component-root"
  >
    <p
      class="MuiTypography-root Component-root Component-root MuiTypography-body2"
    >
      Created
       
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover Component-root Component-root MuiTypography-subtitle2 MuiTypography-colorPrimary"
        href="https://autoseg.com"
      >
        @AutoSeg
      </a>
    </p>
    <p
      class="MuiTypography-root Component-root Component-root MuiTypography-body2"
    >
      Version
       1.0.0
    </p>
    <div
      class="MuiBox-root MuiBox-root Component-root Component-root"
    >
      <button
        aria-label="Portuguese"
        class="MuiButtonBase-root MuiIconButton-root Component-root Component-root Styled(Component)-root Styled(Component)-root WithStyles(ForwardRef(Tooltip))-root WithStyles(ForwardRef(Tooltip))-root"
        data-testid="change-language-portuguese"
        tabindex="0"
        title="Portuguese"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <img
            alt="Brasil flag"
            class="img-root img-root"
            src=""
          />
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="English"
        class="MuiButtonBase-root MuiIconButton-root Component-root Component-root Styled(Component)-root Styled(Component)-root WithStyles(ForwardRef(Tooltip))-root WithStyles(ForwardRef(Tooltip))-root"
        data-testid="change-language-english"
        tabindex="0"
        title="English"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <img
            alt="United Kingdom flag"
            class="img-root img-root"
            src=""
          />
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="Spanish"
        class="MuiButtonBase-root MuiIconButton-root Component-root Component-root Styled(Component)-root Styled(Component)-root WithStyles(ForwardRef(Tooltip))-root WithStyles(ForwardRef(Tooltip))-root"
        data-testid="change-language-spanish"
        tabindex="0"
        title="Spanish"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <img
            alt="Spain flag"
            class="img-root img-root"
            src=""
          />
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`testing Welcome component should snapshot works with portuguese language 1`] = `
<div>
  <div
    class="MuiContainer-root Component-root Component-root MuiContainer-maxWidthMd"
  >
    <img
      alt="Logotipo de bem-vindo"
      class="img-root img-root"
      src=""
    />
    <div
      class="MuiGrid-root WithStyles(ForwardRef(Grid))-root WithStyles(ForwardRef(Grid))-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root WithStyles(ForwardRef(Grid))-root WithStyles(ForwardRef(Grid))-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained Component-root Component-root MuiButton-containedSecondary"
          data-testid="register-button"
          href="/front/register"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            Criar conta
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
      <div
        class="MuiGrid-root WithStyles(ForwardRef(Grid))-root WithStyles(ForwardRef(Grid))-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained Component-root Component-root MuiButton-containedPrimary MuiButton-fullWidth"
          data-testid="login-button"
          href="/front/login"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            Logar
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
    <div
      class="MuiPaper-root Component-root Component-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <p
        class="MuiTypography-root Component-root Component-root MuiTypography-body1 MuiTypography-alignCenter"
        data-testid="welcome-board-title"
      >
        Se você já tem uma conta
      </p>
      <p
        class="MuiTypography-root Component-root Component-root MuiTypography-body1 MuiTypography-alignCenter"
        data-testid="welcome-board-subtitle"
      >
        Logar
      </p>
      <img
        alt="Interface de bem-vindo"
        class="img-root img-root"
        src=""
      />
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root Component-root Component-root"
  >
    <p
      class="MuiTypography-root Component-root Component-root MuiTypography-body2"
    >
      Criado
       
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover Component-root Component-root MuiTypography-subtitle2 MuiTypography-colorPrimary"
        href="https://autoseg.com"
      >
        @AutoSeg
      </a>
    </p>
    <p
      class="MuiTypography-root Component-root Component-root MuiTypography-body2"
    >
      Versão
       1.0.0
    </p>
    <div
      class="MuiBox-root MuiBox-root Component-root Component-root"
    >
      <button
        aria-label="Português"
        class="MuiButtonBase-root MuiIconButton-root Component-root Component-root Styled(Component)-root Styled(Component)-root WithStyles(ForwardRef(Tooltip))-root WithStyles(ForwardRef(Tooltip))-root"
        data-testid="change-language-portuguese"
        tabindex="0"
        title="Português"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <img
            alt="Bandeira do Brasil"
            class="img-root img-root"
            src=""
          />
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="Inglês"
        class="MuiButtonBase-root MuiIconButton-root Component-root Component-root Styled(Component)-root Styled(Component)-root WithStyles(ForwardRef(Tooltip))-root WithStyles(ForwardRef(Tooltip))-root"
        data-testid="change-language-english"
        tabindex="0"
        title="Inglês"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <img
            alt="Bandeira do Reino Unido"
            class="img-root img-root"
            src=""
          />
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="Espanhol"
        class="MuiButtonBase-root MuiIconButton-root Component-root Component-root Styled(Component)-root Styled(Component)-root WithStyles(ForwardRef(Tooltip))-root WithStyles(ForwardRef(Tooltip))-root"
        data-testid="change-language-spanish"
        tabindex="0"
        title="Espanhol"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <img
            alt="Bandeira da Espanha"
            class="img-root img-root"
            src=""
          />
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`testing Welcome component should snapshot works with spanish language 1`] = `
<div>
  <div
    class="MuiContainer-root Component-root Component-root MuiContainer-maxWidthMd"
  >
    <img
      alt="Logotipo de Bienvenido"
      class="img-root img-root"
      src=""
    />
    <div
      class="MuiGrid-root WithStyles(ForwardRef(Grid))-root WithStyles(ForwardRef(Grid))-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root WithStyles(ForwardRef(Grid))-root WithStyles(ForwardRef(Grid))-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained Component-root Component-root MuiButton-containedSecondary"
          data-testid="register-button"
          href="/front/register"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            Crear una cuenta
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
      <div
        class="MuiGrid-root WithStyles(ForwardRef(Grid))-root WithStyles(ForwardRef(Grid))-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained Component-root Component-root MuiButton-containedPrimary MuiButton-fullWidth"
          data-testid="login-button"
          href="/front/login"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            Iniciar sesión
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
    <div
      class="MuiPaper-root Component-root Component-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <p
        class="MuiTypography-root Component-root Component-root MuiTypography-body1 MuiTypography-alignCenter"
        data-testid="welcome-board-title"
      >
        Si ya tienes una cuenta
      </p>
      <p
        class="MuiTypography-root Component-root Component-root MuiTypography-body1 MuiTypography-alignCenter"
        data-testid="welcome-board-subtitle"
      >
        Iniciar sesión
      </p>
      <img
        alt="Interfaz de bienvenido"
        class="img-root img-root"
        src=""
      />
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root Component-root Component-root"
  >
    <p
      class="MuiTypography-root Component-root Component-root MuiTypography-body2"
    >
      Creado
       
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover Component-root Component-root MuiTypography-subtitle2 MuiTypography-colorPrimary"
        href="https://autoseg.com"
      >
        @AutoSeg
      </a>
    </p>
    <p
      class="MuiTypography-root Component-root Component-root MuiTypography-body2"
    >
      Versión
       1.0.0
    </p>
    <div
      class="MuiBox-root MuiBox-root Component-root Component-root"
    >
      <button
        aria-label="Portugués"
        class="MuiButtonBase-root MuiIconButton-root Component-root Component-root Styled(Component)-root Styled(Component)-root WithStyles(ForwardRef(Tooltip))-root WithStyles(ForwardRef(Tooltip))-root"
        data-testid="change-language-portuguese"
        tabindex="0"
        title="Portugués"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <img
            alt="Bandera de Brasil"
            class="img-root img-root"
            src=""
          />
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="Inglés"
        class="MuiButtonBase-root MuiIconButton-root Component-root Component-root Styled(Component)-root Styled(Component)-root WithStyles(ForwardRef(Tooltip))-root WithStyles(ForwardRef(Tooltip))-root"
        data-testid="change-language-english"
        tabindex="0"
        title="Inglés"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <img
            alt="Bandera del Reino Unido"
            class="img-root img-root"
            src=""
          />
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="Español"
        class="MuiButtonBase-root MuiIconButton-root Component-root Component-root Styled(Component)-root Styled(Component)-root WithStyles(ForwardRef(Tooltip))-root WithStyles(ForwardRef(Tooltip))-root"
        data-testid="change-language-spanish"
        tabindex="0"
        title="Español"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <img
            alt="Bandera de España"
            class="img-root img-root"
            src=""
          />
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
